src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetAnnotationExecutor.java

```diff
@@ -49,41 +51,65 @@ public void execLoad(BulkLoadFileHistory bulkLoadFileHistory) {
 			BackendBulkDataProvider dataProvider = BackendBulkDataProvider.valueOf(fms.getFmsDataSubType());
 
 			bulkLoadFileDAO.merge(bulkLoadFileHistory.getBulkLoadFile());
-			
-			List<Long> datasetIdsLoaded = new ArrayList<>();
-			
-			bulkLoadFileHistory.setCount(htpExpressionDatasetData.getData().size());
-			updateHistory(bulkLoadFileHistory);
 
-			boolean result = runLoaddatasetid(externalDataBaseEntityService, bulkLoadFileHistory, dataProvider, htpExpressionDatasetData.getData(), datasetIdsLoaded, false);
+			List<Long> htpAnnotationsIdsLoaded = new ArrayList<>();
+			List<Long> previousIds = htpExpressionDatasetAnnotationService.getAnnotationIdsByDataProvider(dataProvider.name());
+			bulkLoadFileHistory.setTotalRecords((long) htpExpressionDatasetData.getData().size());
+			updateHistory(bulkLoadFileHistory);
+			boolean success = runLoad(bulkLoadFileHistory, dataProvider, htpExpressionDatasetData.getData(), htpAnnotationsIdsLoaded);
+			if(success) {
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.
<!-- __reviewdog__:ChA2NDU0MDJmZjc1YzUxMTMxEgpjaGVja3N0eWxl -->